src/plugins/vis_type_table/public/components/table_vis_basic.tsx

```diff
@@ -0,0 +1,149 @@
+    const sortedRows = useMemo(
+      () =>
+        sort.columnIndex !== null && sort.direction
+          ? orderBy(rows, columns[sort.columnIndex]?.id, sort.direction)
+          : rows,
+      [columns, rows, sort]
+    );
+
+    // renderCellValue is a component which renders a cell based on column and row indexes
+    const renderCellValue = useMemo(() => createTableVisCell(columns, sortedRows), [
+      columns,
+      sortedRows,
+    ]);
+
+    // Columns config
+    const gridColumns = useMemo(() => createGridColumns(table, columns, sortedRows, fireEvent), [
+      table,
+      columns,
+      sortedRows,
+      fireEvent,
+    ]);
+
+    // Pagination config
+    const pagination = usePagination(visConfig);
+    // Sorting config
+    const sortingColumns = useMemo(
+      () =>
+        sort.columnIndex !== null && sort.direction
+          ? [{ id: columns[sort.columnIndex]?.id, direction: sort.direction }]
+          : [],
+      [columns, sort]
+    );
+    const onSort = useCallback(
+      (sortingCols: EuiDataGridSorting['columns'] | []) => {
+        // data table vis sorting now only handles one column sorting
+        // if data grid provides more columns to sort, pick only the next column to sort
+        const newSortValue = sortingCols.length <= 1 ? sortingCols[0] : sortingCols[1];
+        setSort(
+          newSortValue && {
+            columnIndex: columns.findIndex((c) => c.id === newSortValue.id),
+            direction: newSortValue.direction,
+          }
+        );
+      },
+      [columns, setSort]
+    );
+
+    const dataGridAriaLabel =
+      title ||
+      visConfig.title ||
+      i18n.translate('visTypeTable.defaultAriaLabel', {
+        defaultMessage: 'Data table visualization',
+      });
+
+    return (
+      <>
+        {title && (
+          <EuiTitle size="xs">
+            <h3>{title}</h3>
+          </EuiTitle>
+        )}
+        <EuiDataGrid
+          aria-label={dataGridAriaLabel}
+          columns={gridColumns}
+          gridStyle={{
+            border: 'horizontal',
+            header: 'underline',
+          }}
+          rowCount={rows.length}
+          columnVisibility={{
+            visibleColumns: columns.map(({ id }) => id),
+            setVisibleColumns: () => {},
+          }}
+          toolbarVisibility={
+            visConfig.showToolbar && {
+              showColumnSelector: false,
+              showFullScreenSelector: false,
+              showSortSelector: false,
+              additionalControls: (
+                <TableVisControls
+                  dataGridAriaLabel={dataGridAriaLabel}
+                  cols={columns}
+                  rows={rows}
+                  table={table}
+                  filename={visConfig.title}
+                />
+              ),
+            }
+          }
+          renderCellValue={renderCellValue}
+          renderFooterCellValue={
+            visConfig.showTotal
+              ? // @ts-expect-error
```

Comment:
This is concerning - why is this missing in the types? Is this is an oversight or is it intentional and shouldn't be used?

In the first case, please open an issue for EUI, otherwise we should work around this and look up the index by the columnId which seems to get passed in.

cc @chandlerprall

Reply:
I'll create a pr with fixing this on the eui side.